src/server/transaction.cc

```diff
@@ -577,36 +615,23 @@ bool Transaction::RunInShard(EngineShard* shard, bool txq_ooo) {
   return !is_concluding;
 }
 
+// TODO: For multi-transactions we should be able to deduce mode() at run-time based
+// on the context. For regular multi-transactions we can actually inspect all commands.
+// For eval-like transactions - we can decided based on the command flavor (EVAL/EVALRO) or
+// auto-tune based on the static analysis (by identifying commands with hardcoded command names).
 void Transaction::ScheduleInternal() {
   DCHECK(!shard_data_.empty());
   DCHECK_EQ(0u, txid_);
   DCHECK_EQ(0, coordinator_state_ & (COORD_SCHED | COORD_OOO));
+  DCHECK_GT(unique_shard_cnt_, 0);
 
-  bool span_all = IsGlobal();
+  DVLOG(1) << "ScheduleInternal " << cid_->name() << " on " << unique_shard_cnt_ << " shards";
 
-  uint32_t num_shards;
-  std::function<bool(uint32_t)> is_active;
-
-  // TODO: For multi-transactions we should be able to deduce mode() at run-time based
-  // on the context. For regular multi-transactions we can actually inspect all commands.
-  // For eval-like transactions - we can decided based on the command flavor (EVAL/EVALRO) or
-  // auto-tune based on the static analysis (by identifying commands with hardcoded command names).
-  if (span_all) {
-    is_active = [](uint32_t) { return true; };
-    num_shards = shard_set->size();
-  } else {
-    num_shards = unique_shard_cnt_;
-    DCHECK_GT(num_shards, 0u);
-
-    is_active = [&](uint32_t i) {
-      return num_shards == 1 ? (i == unique_shard_id_) : shard_data_[i].local_mask & ACTIVE;
-    };
-  }
```

Comment:
We actually set unique_shard_cnt and ACTIVE flags for a global transaction, so we don't need this code at all